README.md

```diff
@@ -43,25 +43,51 @@ To test, run `make test` and to see code coverage, run `make cover`.
-other source; right now, mitmengine supports reading these files from Amazon S3 client-compatible databases (including 
-Amazon S3 and Ceph). Additional file readers for databases (which we call "loaders") can be defined in the `loaders` 
-package, and as long as new loaders implement the Loader interface, they should work with the rest of mitmengine out of the 
-box. We added support for additional fingerprint and bad header sources in the case mitmengine is run as a daemon and 
+headers. Alternatively, it can also specify a configuration file for reading the previously mentioned files from any
+other source; right now, mitmengine supports reading these files from Amazon S3 client-compatible databases (including
+Amazon S3 and Ceph). Additional file readers for databases (which we call "loaders") can be defined in the `loaders`
+package, and as long as new loaders implement the Loader interface, they should work with the rest of mitmengine out of the
+box. We added support for additional fingerprint and bad header sources in the case mitmengine is run as a daemon and
 you want to have it periodically update the fingerprint and bad header files it uses to analyze traffic.
 
 The intended entrypoint to the mitmengine package is through the `Processor.Check` function, which takes a user agent and client request fingerprint, and returns a mitm detection report. Additional API functions will be added in the future to allow for adding new signatures to a running process, for example.
 
 ## How to Contribute
 
-As browser and middleware fingerprints quickly become outdated, we are actively seeking to update the fingerprint repository with new samples. To contribute fingerprint samples, please submit a PR with the following:
-- Pcap (generated by `tcpdump` or `wireshark`, for example) containing a Client Hello generated by the TLS client to fingerprint at `testdata/pcaps/<desc>/handshake.pcap`, where `<desc>` is a unique and descriptive name for the pcap sample
-- (Optional) Pcap containing an unencrypted HTTP request generated by the HTTP client to fingerprint at `testdata/pcaps/<desc>/header.pcap`
-- Line added to `testdata/fingerprint_metadata.jsonl` with the below fields. Recognized options for the `os`, `device`, `platform`, and `browser` fields are those defined in the `uasurfer` package. Recognized options for `middleware_fingerprint.type` are listed below. See `testdata/fingerprint_metadata.jsonl` for examples; any unknown fields can be left blank or omitted.
+As browser and middleware fingerprints quickly become outdated, we are actively seeking to update the fingerprint repository with new samples. To contribute fingerprint samples, please follow these steps:
+
+### Generate a fingerprint sample
+(tested on macOS Mojave 10.14.3)
+
+- Create server RSA certificate and key pair:
+```
+openssl req -new -x509 -out server.crt -nodes -keyout server.pem -subj /CN=localhost`
```

Comment:
This and the following three commands have an excess trailing backtick.

Maybe you should add a `-sha256` for good measure, in case clients start refusing SHA-1 certificates?

Reply:
Done, thanks!